lib/search_works/links.rb

```diff
@@ -39,26 +39,38 @@ def ill
     end
 
     class Link
-      attr_accessor :html, :text, :href, :file_id, :druid, :type, :sort
+      include ActionView::Helpers::TagHelper
+
+      attr_accessor :href, :file_id, :druid, :type, :sort
 
       def initialize(options = {})
-        @html = options[:html]
-        @text = options[:text]
-        @href = options[:href]
-        @fulltext = options[:fulltext]
-        @stanford_only = options[:stanford_only]
-        @finding_aid = options[:finding_aid]
-        @sfx = options[:sfx]
-        @managed_purl = options[:managed_purl]
-        @file_id = options[:file_id]
+        @additional_text = options[:additional_text]
+        @casalini = options[:casalini]
         @druid = options[:druid]
+        @file_id = options[:file_id]
+        @finding_aid = options[:finding_aid]
+        @fulltext = options[:fulltext]
+        @href = options[:href]
         @ill = options[:ill]
-        @type = options[:type]
+        @link_text = options[:link_text]
+        @link_title = options[:link_title]
+        @managed_purl = options[:managed_purl]
+        @sfx = options[:sfx]
         @sort = options[:sort]
+        @stanford_only = options[:stanford_only]
+        @type = options[:type]
       end
 
       def ==(other)
-        [@html, @fulltext, @stanford_only, @finding_aid, @sfx] == [other.html, other.fulltext?, other.stanford_only?, other.finding_aid?, other.sfx?]
+        [@href, @fulltext, @stanford_only, @finding_aid, @sfx] == [other.href, other.fulltext?, other.stanford_only?, other.finding_aid?, other.sfx?]
+      end
+
+      def html
+        @html ||= [link_html, casalini_text, additional_text_html].compact.join(' ').strip
```

Comment:
```suggestion
        @html ||= safe_join [link_html, casalini_text, additional_text_html].compact.map(&:strip), ' '
```

I think the regular join is going to lose its html safety otherwise?

Reply:
you're right...and strip does the same